plugins/welcome-plugin/src/welcome-page.ts

```diff
@@ -0,0 +1,209 @@
+
+import * as theia from '@theia/plugin';
+import * as path from 'path';
+
+/**
+ * Welcome Page
+ */
+export class WelcomePage {
+    static readonly ECLIPSE_CHE = 'Eclipse Che';
+    static readonly ECLIPSE_CHE_SUBTITLE = 'Welcome To Your Cloud Developer Workspace ';
+
+    static readonly DOCUMENTATION = 'https://www.eclipse.org/che/docs/che-7';
+    static readonly MATTERMOST = 'https://mattermost.eclipse.org/eclipse/channels/eclipse-che';
+
+
+
+    constructor(readonly pluginContext: theia.PluginContext) {
+
+    }
+
+    protected renderHeader(context: theia.PluginContext): string {
+
+        // Local path to main script run in the webview
+        const imgOnDisk = theia.Uri.file(path.join(context.extensionPath, 'resources', 'che-logo.svg'));
+        // And the uri we use to load this script in the webview
+        const imgURI = imgOnDisk.with({ scheme: 'theia-resource' });
+
+        return `<div class="che-welcome-header">
+         <div class="che-welcome-header-title" ><div class="svg-container"><img src=${imgURI} /></div>${WelcomePage.ECLIPSE_CHE}</div>
+        <span class='che-welcome-header-subtitle'>${WelcomePage.ECLIPSE_CHE_SUBTITLE}</span>
+    </div>`;
+    }
+
+    private async renderCommandKeyBinding(commandId: string): Promise<string> {
+        const availableKeys = await theia.commands.getKeyBinding(commandId);
+
+        if (availableKeys && availableKeys.length > 0) {
+            const keybindingSeperator = /<match>\+<\/match>/g;
+            const regex = new RegExp(keybindingSeperator);
+            const keyMappings = await Promise.all(availableKeys.map(async (keyBinding: any) => {
+
+                const updatedKeyBinding = keyBinding.value.replace(regex, '+');
+                const keys = updatedKeyBinding.split('+');
+                if (keys.length > 0) {
+                    let rows: any[] = [];
+
+                    await Promise.all(keys.map(async (key: any) => {
+                        let updatedKey = key;
+                        if ((await theia.env.getClientOperatingSystem()) === theia.OperatingSystem.OSX) {
+                            if (updatedKey === 'ctrlcmd') {
+                                updatedKey = '⌘ cmd';
+                            } else if (updatedKey === 'alt') {
+                                updatedKey = '⎇ alt';
+                            }
+                        }
+                        if (updatedKey === 'shift') {
+                            updatedKey = '⇧ shift';
+                        }
+                        rows.push(`<span class="che-welcome-keybinding-key">${updatedKey}</span>`);
+                    }));
+                    return `<div class="che-welcome-keybinding" title=${availableKeys[0].value}>${rows}</div>`;
+                }
+            }));
+
+            return keyMappings.join(' or ')
+
+
+
+        }
+        return '';
+    }
+
+
+    protected async renderCommandKeyBindingOld(commandId: string): Promise<string> {
```

Comment:
It seems like this method isn't used and could be removed.

Reply:
yes good catch @akurinnoy